solution/index.js

```diff
@@ -0,0 +1,232 @@
+        }
+      }
+    }
+  }
+  
+  sections.addEventListener('dblclick', changeTask)
+  
+  function changeTask(e) {
+    e.preventDefault()
+    const target = e.target
+    const saveKey = tasksObj[target.closest('ul').id] //now saveKey contains for example: taskObj.todo
+    const oldText = target.textContent
+    target.setAttribute('contentEditable', 'true')
+    target.addEventListener('blur', () => {
+      let newText = target.textContent
+      if (newText === '') target.textContent = oldText //bug fixed - when we dblclick and then remove the content and then lose focus so the old content came up and the task wouldn't stay empty
+      newText = target.textContent
+      saveKey[saveKey.findIndex((a) => a === oldText)] = newText
+      localStorage.setItem('tasks', JSON.stringify(tasksObj))
+    })
+  }
+  
+  
+  /**
+   * Creates a new DOM element.
+   *
+   * Example usage:
+   * createElement("div", ["just text", createElement(...)], ["nana", "banana"], {id: "bla"}, {click: (...) => {...}})
+   *
+   * @param {String} tagName - the type of the element
+   * @param {Array} children - the child elements for the new element.
+   *                           Each child can be a DOM element, or a string (if you just want a text element).
+   * @param {Array} classes - the class list of the new element
+   * @param {Object} attributes - the attributes for the new element
+   */
+  function createElement(
+    tagName,
+    children = [],
+    classes = [],
+    attributes = {},
+    eventListeners = {}
+  ) {
+    let element = document.createElement(tagName)
+  
+    for (const child of children) {
+      element.append(child)
+    }
+  
+    for (const cls of classes) {
+      element.classList.add(cls)
+    }
+  
+    for (const attr in attributes) {
+      element.setAttribute(attr, attributes[attr])
+    }
+    for (const listener in eventListeners) {
+      const functionArray = eventListeners[listener]
+      element.addEventListener(listener, functionArray)
+    }
+  
+    return element
+  }
+  
+  function generateTasks() {
+    //creates tasks to the page, meant for the case that we refresh the page to save all the existed tasks
+    const ulPogress = document.getElementById("in-progress")
+    
+    for (let task of tasksObj.todo) {
+      let li = createElement('li', [], ['task'], { tabindex: '0' })
+      li.innerHTML = task
+      todo.append(li)
+    }
+    for (let task of tasksObj['in-progress']) {
+      let li = createElement('li', [], ['task'], { tabindex: '0' })
+      li.innerHTML = task
+      ulPogress.append(li) //pay attention - here we want to get to the ul with id : in-progress children[i], i is is affected by the ul tag placement
+    }
+    for (let task of tasksObj.done) {
+      let li = createElement('li', [], ['task'], { tabindex: '0' })
+      li.innerHTML = task
+      done.append(li)
+    }
+  }
+  
+  sections.addEventListener('keydown', moveTask)
+  function moveTask(event) {
```

Comment:
There is a lot of code duplication in this function, and I would split the actual moving logic away from the event handling logic.